### ISSUE TITLE
False positive: `Type` objects (including `type[cls]`) are incompatible with `Hashable`

### ISSUE THREAD

**Issue by aatle** [python/mypy#17730]
**Bug Report**
In python, classes themselves, like `int`, are hashable.
A variable annotated with just `type` is correctly recognized as an instance of `collections.abc.Hashable`; however, a variable annotated with `type[int]`, `Type`, or `Type[int]` is wrongly considered incompatible with `Hashable`.

**To Reproduce**

```python
from collections.abc import Hashable
from typing import Type, reveal_type

a: type = int
b: type[int] = int
c: Type = int

x: Hashable = a
y: Hashable = b
z: Hashable = c

reveal_type(a)
reveal_type(b)
reveal_type(c)
```

**Expected Behavior**

No error, and consistent among `x`, `y`, and `z`.

**Actual Behavior**
Both `y` and `z` give an error, but not `x`.
```
main.py:9: error: Incompatible types in assignment (expression has type "type[int]", variable has type "Hashable")  [assignment]
main.py:10: error: Incompatible types in assignment (expression has type "type[Any]", variable has type "Hashable")  [assignment]
main.py:12: note: Revealed type is "builtins.type"
main.py:13: note: Revealed type is "type[builtins.int]"
main.py:14: note: Revealed type is "type[Any]"
```
Note: `type` and `Type` are considered different: the problem lies with `Type`.
**Your Environment**

- Mypy version used: 1.11.2
- Mypy command-line flags: default playground
- Python version used: 3.12

**Comment by chadrik** [comment]
This duplicates https://github.com/python/mypy/issues/11470
